### DISCUSSION TITLE
Adding a 3D object to the map

### DISCUSSION THREAD

**Discussion by frobones** [heremaps/harp.gl#2034] [Q&A] [answered]
Hi all,

I hope this is the right place for something like this. I'm trying to add a 3D object to my map following the tutorial: https://developer.here.com/tutorials/harpgl/#add-3d-objects-to-the-map. However, `cube` as a `THREE.Mesh` neither has a `.geoPosition` or a `.anchor` field. Looking at the page source for https://www.harp.gl/docs/master/examples/#threejs_add-object.html, I see `cube` is actually a `MapAnchor<THREE.Object3D>`.

Am I doing something wrong with following the tutorial?

**Comment by FraukeF** [comment, marked as answer]
Hi @frobones ,

you are at the right place :)

The tutorial you mentioned above is not using types wherefor the code works without complains if adding additional properties like 'anchor' to a THREE.Mesh.
The missing properties are as you already saw coming from the MapAnchor Type.

You can type your cube object to solve the typing this:


```


           const cube: MapAnchor = new THREE.Mesh(geometry, material);

            cube.anchor = geoPosition;

            mapView.mapAnchors.add(cube);
```

**Reply by nzjony** [reply]
@frobones , thanks for the question and thanks @FraukeF  for answering. I will update the tutorial to make this clearer for people who may be stuck in future.

**Reply by frobones** [reply]
@FraukeF Awesome, thank you for the help.

**Reply by nzjony** [reply]
@frobones  I updated the docs: https://developer.here.com/tutorials/harpgl/#add-3d-objects-to-the-map and also found a minor issue as well.
